Demote int64 constant value to int32

sshonTT opened the pull request:
### Ticket
[Link to Github Issue
](https://github.com/tenstorrent/tt-torch/issues/663)

### Problem description
Our device misinterprets int64 tensors when they are provided as constants, resulting in like [1, 0, 2, 0, 3, 0, ...] instead of [1, 2, 3, ...]. 

### What's changed
To prevent this, this function safely casts int64 tensors to int32 if all values fit within the int32 range, ensuring correct data is passed to the device.

### Checklist
- [v] Run swin_b without PCC drops with https://github.com/tenstorrent/tt-mlir/pull/3320

AleksKnezevic commented on tt_torch/dynamo/passes.py:
```diff
@@ -110,6 +110,27 @@ def constant_fold(gm):
+"""
+
+
+def sanitize_int64_constants_for_device(constant_inputs):
```
We already have infra for this, see `typecast_inputs` in executor.py. We should add this to that.

sshonTT replied:
Right, I had a discussion with Defne about this. Thanks, I will check.

sshonTT replied:
I was able to fix it by using `typecast_inputs`.

AleksKnezevic replied:
How about you move the call to typecast just after line 316 in [executor.py](https://github.com/tenstorrent/tt-torch/blob/5a3289e89febc726b2da28016362da13f4048621/tt_torch/dynamo/executor.py#L316). Then it's all in one place.

sshonTT replied:
That makes sense. I was also thinking about how to handle `self.compiler_config.typecast_inputs`. Since it appears to be enabled by default, I think I can go ahead and move it.

sshonTT replied:
Fixed it.

sshonTT committed:
Enable typecast to graph constants

Workaround for int64 constant feeding issue on our device.
Our device misinterprets int64 tensors when they are provided as constants, resulting in like [1, 0, 2, 0, 3, 0, ...] instead of [1, 2, 3, ...].
To prevent this, this function safely casts int64 tensors to int32 if all values fit within the int32 range, ensuring correct data is passed to the device.
